src/Engine/McBopomofoLM.cpp

```diff
@@ -122,7 +124,31 @@ std::vector<Formosa::Gramambular2::LanguageModel::Unigram> McBopomofoLM::getUnig
         allUnigrams = filterAndTransformUnigrams(rawGlobalUnigrams, excludedValues, insertedValues);
     }
 
-    allUnigrams.insert(allUnigrams.begin(), userUnigrams.begin(), userUnigrams.end());
+    // This relies on the fact that we always use the default separator.
+    bool isKeyMultiSyllable = key.find(Formosa::Gramambular2::ReadingGrid::kDefaultSeparator) != std::string::npos;
+    if (isKeyMultiSyllable || allUnigrams.empty()) {
```

Comment:
I guess we may need to comment on why multi syllables phrases should be always in the front without scoring. I cannot understand its purpose when I read the code.

Reply:
Clarified the motivation in 4d9db9ccf075487c86c15885f0086e056f8076d1. PTAL.